src/esp/gfx/replay/Player.cpp

```diff
@@ -239,16 +264,12 @@ void Player::applyKeyframe(const Keyframe& keyframe) {
       implementation_->deleteAssetInstance(it->second);
       createdInstances_.erase(deletionInstanceKey);
     }
-    // The batch renderer can only clear the scene entirely; it cannot delete
-    // individual objects. To process deletions, all instances are deleted,
-    // remaining instances are re-created and latest transform updates are
-    // re-applied.
   } else if (keyframe.deletions.size() > 0) {
     // Cache latest transforms
-    std::unordered_map<RenderAssetInstanceKey, Mn::Matrix4> latestTransforms{};
     for (const auto& pair : this->createdInstances_) {
       const RenderAssetInstanceKey key = pair.first;
-      latestTransforms[key] = implementation_->getNodeTransform(pair.second);
+      latestTransformCache_[key] =
```

Comment:
You should clear this cache before every use so it doesn't grow and grow over time.